client/src/pages/private/ParticipantTable.js

```diff
@@ -0,0 +1,799 @@
+                        setAnchorElement(event.currentTarget);
+                      }}
+                      variant="outlined"
+                      size="small"
+                      text="Actions"
+                    />
+                  }
+                  if (columnId === 'edit') {
+                    return <Button
+                      onClick={async () => {
+                        // Get data from row.id
+                        const response = await fetch(`/api/v1/participant?id=${row.id}`, {
+                          headers: {
+                            'Accept': 'application/json',
+                            'Content-type': 'application/json',
+                            'Authorization': `Bearer ${store.get('TOKEN')}`,
+                          },
+                          method: 'GET',
+                        });
+
+                        const participant = await response.json();
+
+                        setActionMenuParticipant(participant[0]);
+                        setActiveModalForm('edit-participant');
+                        setAnchorElement(null);
+                      }}
+                      variant="outlined"
+                      size="small"
+                      text="Edit"
+                    />
+                  }
+                  if (columnId === 'updated_at') {
+                    return fuzzyDateOffset(row.engage.updated_at);
+                  }
+                  return row[columnId];
+                }
+              }
+              onRequestSort={handleRequestSort}
+              rows={rows}
+              isLoading={isLoadingData}
+            />
+          </Box>
+        </Grid>
+        {(!roles.includes('ministry_of_health')) && (!roles.includes('superuser')) && <Menu
+          keepMounted
+          open={actionMenuParticipant != null && activeModalForm == null}
+          anchorEl={anchorElement}
+          onClose={() => setActionMenuParticipant(null)}
+        >
+          {actionMenuParticipant?.status === 'open' && <MenuItem onClick={() => handleEngage(actionMenuParticipant.id, 'prospecting')}>Engage</MenuItem>}
+          {actionMenuParticipant?.status === 'prospecting' && <MenuItem onClick={() => setActiveModalForm('interviewing')}>Interviewing</MenuItem>}
+          {actionMenuParticipant?.status === 'interviewing' && <MenuItem onClick={() => handleEngage(actionMenuParticipant.id, 'offer_made')}>Offer Made</MenuItem>}
+          {actionMenuParticipant?.status === 'offer_made' && <MenuItem onClick={() => setActiveModalForm('hired')}>Hire</MenuItem>}
+          {['prospecting', 'interviewing', 'offer_made'].includes(actionMenuParticipant?.status) && <MenuItem onClick={() => setActiveModalForm('rejected')}>Archive</MenuItem>}
+          {actionMenuParticipant?.status === 'rejected' && <MenuItem onClick={() => handleEngage(actionMenuParticipant.id, 'prospecting')}>Re-engage</MenuItem>}
+        </Menu>
+        }
+      </CheckPermissions>
+    </>
+  );
+};
```

Comment:
So the reasoning here is to relocate the majority of the participant view logic to this file? We've gone from one 850LOC file to 800LOC and 50LOC files. Probably a very small gain, but a start I suppose. This may be further addressed by HCAP-351.

Reply:
I think I see what you were doing. This is following a pattern of keeping tables separate from the views they're included in. Also done with sites view.

Reply:
Yes, since the sites table needs to be accessed in more than one way, I figured it best to encapsulate it. I did the same with the participants table for the sake of consistency.